File: backend/app/models/pvp_season.py
from typing import Any
from datetime import datetime
from sqlmodel import SQLModel, Field, Column, JSON, UniqueConstraint

class PvPSeason(SQLModel, table=True):
    __tablename__ = "pvp_seasons"

    id: int = Field(primary_key=True, index=True)
    name_json: dict[str, Any] = Field(
        default_factory=dict,
        sa_column=Column(JSON),
        description="Localized names for the PvP season",
    )
    is_current: bool = Field(default=False, index=True)
    updated_at: datetime = Field(default_factory=datetime.utcnow, index=True)
    raw_json: dict[str, Any] = Field(
        default_factory=dict,
        sa_column=Column(JSON),
        description="Raw JSON data from the API",
    )

    __table_args__ = (
        UniqueConstraint("id", name="uq_pvp_seasons_id"),
    )

File: backend/app/models/pvp_season.rb
# == Schema Information
#
# Table name: pvp_seasons
#
#  id           :bigint           not null, primary key
#  display_name :string
#  end_time     :datetime
#  is_current   :boolean          default(FALSE)
#  start_time   :datetime
#  created_at   :datetime         not null
#  updated_at   :datetime         not null
#  blizzard_id  :string
#
# Indexes
#
#  index_pvp_seasons_on_is_current  (is_current)
#  index_pvp_seasons_on_updated_at  (updated_at)
#
class PvpSeason < ApplicationRecord
end
